File: fe/src/services/api/routes/marketplace.route.js
import api from "../axios";
export const marketplaceAPI = {
  getAll: (params) => api.get("/marketplace", { params }),
  getById: (id) => api.get(`/marketplace/${id}`),
  create: (data, fotoBarangFiles) => {
    if (fotoBarangFiles) {
      const formData = new FormData();
      fotoBarangFiles.foto_barang_urls.forEach((file) => {
        formData.append("foto_barang_urls", file);
      });
      Object.entries(data).forEach(([key, value]) => {
        if (value !== "" && value !== null && value !== undefined) {
          if (Array.isArray(value)) {
            value.forEach((val) => formData.append(key, val));
          } else {
            formData.append(key, value);
          }
        }
      });
      return api.post("/marketplace", formData, {
        headers: { "Content-Type": "multipart/form-data" },
      });
    }
    return api.post("/marketplace", data);
  },
  update: (id, data, fotoBarangFiles = null) => {
    if (fotoBarangFiles || (data && Array.isArray(data.existing_pictures))) {
      const formData = new FormData();
      if (fotoBarangFiles && fotoBarangFiles.foto_barang_urls) {
        fotoBarangFiles.foto_barang_urls.forEach((file) => {
          formData.append("foto_barang_urls", file);
        });
      }
      Object.entries(data).forEach(([key, value]) => {
        if (value !== "" && value !== null && value !== undefined) {
          if (Array.isArray(value)) {
            value.forEach((val) => formData.append(key, val));
          } else {
            formData.append(key, value);
          }
        }
      });
      return api.put(`/marketplace/${id}`, formData, {
        headers: { "Content-Type": "multipart/form-data" },
      });
    }
    return api.put(`/marketplace/${id}`, data);
  },
  delete: (id) => api.delete(`/marketplace/${id}`),
  updateKetersediaan: (id, data) =>
    api.patch(`/marketplace/${id}/update-ketersediaan`, data),
  getMyMarketplace: (params) =>
    api.get("/marketplace/my-marketplace", { params }),
};

File: be/app/api/routes/marketplace_route.py
"""Marketplace routes"""
from flask import Blueprint
from app.api.controllers import marketplace_controller

marketplace_bp = Blueprint('marketplace', __name__, url_prefix='/api/marketplace')


@marketplace_bp.route('', methods=['GET'])
def get_all():
    return marketplace_controller.get_all()


@marketplace_bp.route('/<item_id>', methods=['GET'])
def get_one(item_id):
    return marketplace_controller.get_one(item_id)


@marketplace_bp.route('', methods=['POST'])
def create():
    return marketplace_controller.create()


@marketplace_bp.route('/<item_id>', methods=['PUT', 'PATCH'])
def update(item_id):
    return marketplace_controller.update(item_id)


@marketplace_bp.route('/<item_id>', methods=['DELETE'])
def delete(item_id):
    return marketplace_controller.delete(item_id)


@marketplace_bp.route('/<item_id>/update-ketersediaan', methods=['PATCH'])
def update_ketersediaan(item_id):
    return marketplace_controller.update_ketersediaan(item_id)


@marketplace_bp.route('/my-marketplace', methods=['GET'])
def my_marketplace():
    return marketplace_controller.my_marketplace()
